cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -65,14 +65,14 @@
   private final ExploreFacade exploreFacade;
   private final boolean allowDatasetUncheckedUpgrade;
   private final UsageRegistry usageRegistry;
-  private final AbstractNamespaceClient namespaceClient;
+  private final NamespaceStore nsStore;
```

Comment:
It might not be a good idea to use NamespaceStore directly. Ideally we should be using NamespaceAdmin interface here.